### PR TITLE
SPLAT-1272: Update for Nutanix failure domains api changes

### PR DESCRIPTION
**Pull request by yanhua121**
Update for Nutanix failure domains api changes.
Depends on the api PR: https://github.com/openshift/api/pull/1578

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -259,3 +259,8 @@ require (
 	sigs.k8s.io/kustomize/kyaml v0.14.3-0.20230601165947-6ce0bf390ce3 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
+
+// FIXME: For Nutanix failure domain support
+replace github.com/openshift/api => github.com/nutanix-cloud-native/openshift-api v0.0.0-20231115190355-c3972acd2be6
+
+replace github.com/openshift/client-go => github.com/yanhua121/client-go v0.0.0-20231115191314-2d5bcdc083eb
```
Comment (elmiko) [inline go.mod:266]: i'm guessing we are waiting for the other PRs to merge?

**Review by elmiko — COMMENTED** [review]
looks good to me, just need to fix the replace statements when the others merge.

File: go.mod (hunk 1 above)
Comment (yanhua121) [reply go.mod]: Updated the dependency versioning.

**Comment by JoelSpeed** [comment]
/label acknowledge-critical-fixes-only

This should be a benign change, no concern about breaking things here if the CI is stable

**Pull request merged**